client/castApp/src/caster.c

```diff
@@ -114,9 +116,18 @@ void casterInit(caster_t *self)
     self->onmsg = &casterShowMsgDefault;
     self->current = casterStateInit;
     self->timeout = reccastTimeout;
-    ellInit(&self->extra_envs);
+    ellInit(&self->envs);
     ellInit(&self->exclude_patterns);
 
+    /* add default_envs to envs list which can be expanded by the user with addReccasterEnvVars iocsh function */
+    for (i = 0; default_envs[i]; i++) {
+        const size_t arg_len = strlen(default_envs[i]) + 1;
+        string_list_t *new_node = mallocMustSucceed(sizeof(string_list_t) + arg_len, "casterInit");
+        new_node->item_str = (char *)(new_node + 1);
+        memcpy(new_node->item_str, default_envs[i], arg_len);
+        ellAdd(&self->envs, &new_node->node);
+    }
```

Comment:
Can you not just replace this with a call to
```c
addToReccasterLinkedList(self, num_default_envs, default_envs, self->envs, "casterInit", "default_envs");
```
or something like that?

Reply:
Yes that makes it cleaner. I removed the `itemCount` argument from `addToReccasterLinkedList` since argv and default_envs are both null terminated arrays. It is easy to switch back to use argc and then count the num_default_envs if you think that is better

Reply:
also once the default_env handling was switched to use `addToReccasterLinkedList` , I noticed warnings that there is a duplicate env var `HOSTNAME`. Removed the extra `HOSTNAME` definition in dbcb.c which I don't think will impact anything but thought worth mentioning